client/core/types.go

```diff
@@ -422,6 +421,13 @@ func coreOrderFromTrade(ord order.Order, metaData *db.OrderMetaData) *Order {
 		AccelerationCoins: accelerationCoins,
 	}
 
+	if !temp {
+		corder.Stamp = uint64(prefix.ServerTime.UnixMilli())
+		corder.ID = ord.ID().Bytes()
+	} else {
+		corder.ID = encode.RandomBytes(order.OrderIDSize)
```

Comment:
Related to the other comment about not having `coreOrderFromTrade` generate the random ID (it needs to be a deterministic function), it's not good to have the `ID` field change when the order goes from pending -> concrete.

How come `coreOrderFromTrade` leaves the `TempID` field zero?